File: C/LinearSearch.c
#include <stdlib.h>

int HasLinearWStart(int searched,int arr[],int size,int start){
	for(int i = start;i < size; i++)
		if(arr[i] == searched)
			return 1;
	return 0;
}
int HasLinear(int searched,int arr[],int size){
	return HasLinearWStart( searched, arr, size, 0);
}
int PosLinearWStart(int searched,int arr[],int size,int start){
	for(int i = start;i < size; i++)
		if(arr[i] == searched)
			return i;
	return -1;
}
int PosLinear(int searched,int arr[],int size){
	return PosLinearWStart( searched, arr, size, 0);
}


int main(){
}

File: Java/LinearSearch.java

public class LinearSearch 
{
	public static int PosLinear(int searched, int[] arr, int size, int start)
	{
		for(int i = start;i < size; i++)
		{
			if(searched == arr[i])
			{
				return i;
			}
		}
		return -1;
	}
	
	public static int PosLinear(int searched, int[] arr, int size)
	{
		return PosLinear(searched,arr,size,0);
	}
	public static boolean HasLinear(int searched, int[] arr, int size, int start)
	{
		for(int i = start;i < size; i++)
		{
			if(searched == arr[i])
			{
				return true;
			}
		}
		return false;
	}
	
	public static boolean HasLinear(int searched, int[] arr, int size)
	{
		return HasLinear(searched,arr,size,0);
	}
}
